plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -141,7 +250,62 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 				);
 			}
 		}
+	}
 
+	/**
+	 * Adds a preload link for a <picture> element.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_Tag_Visitor_Context                                                                                                                    $context Tag visitor context.
+	 * @param string                                                                                                                                    $xpath   XPath of the element.
+	 * @param array{srcset?: string|true|null,sizes?: string|true|null,type?: string|true|null,media?: string|true|null,crossorigin?: string|true|null} $source  Collected sources from the <source> elements.
+	 */
+	private function add_preload_link_for_picture( OD_Tag_Visitor_Context $context, string $xpath, array $source ): void {
+		// If this element is the LCP (for a breakpoint group), add a preload link for it.
+		foreach ( $context->url_metric_group_collection->get_groups_by_lcp_element( $xpath ) as $group ) {
+			$link_attributes = array_merge(
+				array(
+					'rel'           => 'preload',
+					'fetchpriority' => 'high',
+					'as'            => 'image',
+				),
+				array_filter(
+					array(
+						'href'        => isset( $source['srcset'] ) && is_string( $source['srcset'] )
+						? explode( ' ', $source['srcset'] )[0]
+						: '',
+						'imagesrcset' => isset( $source['srcset'] ) && is_string( $source['srcset'] ) ? $source['srcset'] : '',
+						'imagesizes'  => isset( $source['sizes'] ) && is_string( $source['sizes'] ) ? $source['sizes'] : '',
+						'type'        => isset( $source['type'] ) && is_string( $source['type'] ) ? $source['type'] : '',
+						'media'       => isset( $source['media'] ) && is_string( $source['media'] ) ? 'screen and ' . $source['media'] : 'screen',
+					),
+					static function ( string $value ): bool {
+						return '' !== $value;
+					}
+				)
+			);
+			if ( isset( $source['crossorigin'] ) ) {
+				$link_attributes['crossorigin'] = 'use-credentials' === $source['crossorigin'] ? 'use-credentials' : 'anonymous';
+			}
+			$context->link_collection->add_link(
+				$link_attributes,
+				$group->get_minimum_viewport_width(),
+				$group->get_maximum_viewport_width()
+			);
+		}
+	}
+
+	/**
+	 * Adds a preload link for an <img> element.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_HTML_Tag_Processor  $processor HTML tag processor.
+	 * @param OD_Tag_Visitor_Context $context   Tag visitor context.
+	 * @param string                 $xpath     XPath of the element.
+	 */
+	private function add_preload_link_for_img( OD_HTML_Tag_Processor $processor, OD_Tag_Visitor_Context $context, string $xpath ): void {
```

Comment:
Per above, I think this logic can remain in `process_img` but just to wrap it in a check like `if ( 'PICTURE' !== $parent_tag )`.

Also, I think this logic could be put inside `process_img()`, unless there is an opportunity to refactor it so that it can be reused both for `PICTURE` and `IMG`.